### ISSUE TITLE
Feature Request: Add `poll_events` API to integrate with other event loop

### ISSUE THREAD

**Issue by zuoxinyu** [slint-ui/slint#2763]
I'm using C++ API, need a `slint::poll_events` API to integrate with other event system like SDL2 or libuv. Currently there is a workaround by using a repeat timer. But I'm glad to see there would be a dedicate API for this purpose.

**Comment by tronical** [comment]
Yes, perhaps this is doable. What's a little tricky is that neither with Qt nor with Winit we can get the file descriptors needed for polling. Could you describe a little how your ideal setup would look like?

**Comment by zuoxinyu** [comment]
> Yes, perhaps this is doable. What's a little tricky is that neither with Qt nor with Winit we can get the file descriptors needed for polling. Could you describe a little how your ideal setup would look like?

I don't think it has anything to do with file descriptors here. 

A `poll()` API typically is used as ` while (true) { poll(); } ` instead of `slint::run_event_loop()`. Inner the loop users can do some other stuffs.

**Comment by tronical** [comment]
Right, so you'd basically run a while loop that first polls Slint, then polls SDL, then maybe libuv - each with a timeout one after the other? It's not optimal, but probably acceptable for many :-)

We could offer this, indeed - using processEvents() with Qt and `run_return()` with winit - despite all the [caveats](https://docs.rs/winit/latest/winit/platform/run_return/trait.EventLoopExtRunReturn.html#caveats).

**Comment by zuoxinyu** [comment]
Yes, users who use this api should take care about the caveats themeselves.

**Comment by zuoxinyu** [comment]
> > These are very common usecases in an eventloop based programming model. In other frameworks, an event loop is always combined with a poll semantic such as [SDL2](https://wiki.libsdl.org/SDL2/SDL_PollEvent), [libuv](http://docs.libuv.org/en/v1.x/loop.html#c.uv_run), [Qt](https://doc.qt.io/qt-6/qeventloop.html#processEvents), [asio](https://www.boost.org/doc/libs/1_82_0/doc/html/boost_asio/reference/io_context/poll.html).
> 
> This is specifically incompatible with target environments where the platform owns the event loop. This includes every web browser, Windows, MacOS, and both major mobile platforms. `winit` was originally built the way you describe, with the user running a `loop {}` in `fn main()`, but ultimately [was rearchitected](https://github.com/rust-windowing/winit/issues/459) to invert the control flow. Having the platform loop call the application _always_ works, while having an application loop call the platform doesn't.
> 
> Slint could offer a `poll_events()` API in at least some configurations, but I don't think it's viable to offer it in all configurations.

Then alternatively, Slint could provide a hook method to support this.

**Comment by ogoffart** [comment]
Maybe it would be useful to understand the exact use-case and how we can solve it.
What are we trying to solve?
Can this be solved by using thread and callbacks? (`slint::invoke_from_event_loop` or `slint::blocking_invoke_from_event_loop`
